helios-ts/src/storage.rs

```
@@ -0,0 +1,45 @@
+extern crate console_error_panic_hook;
+extern crate web_sys;
+
+use config::Config;
+use consensus::database::Database;
+use eyre::Result;
+
+#[derive(Clone)]
+pub struct StorageDB {
+    checkpoint: Vec<u8>,
+}
+
+impl Database for StorageDB {
+    fn new(config: &Config) -> Result<Self> {
+        console_error_panic_hook::set_once();
+        let window = web_sys::window().unwrap();
+        if let Ok(Some(local_storage)) = window.local_storage() {
+            let checkpoint = local_storage.get_item("checkpoint").unwrap();
```

Lets return an err result if it fails here.